src/main/javascript/util/MslUtils.js

```diff
@@ -42,7 +42,8 @@ var MslUtils$uncompress;
         case CompressionAlgorithm.LZW:
             return lzw$compress(data);
         case CompressionAlgorithm.GZIP:
-            return gzip$compress(data);
+            if (gzip$compress)
```

Comment:
What is `gzip$compress` and where is it defined? What are its possible values? If the consumer of the MSL API is expected to define it, would it be better to inject it so it's not a floating global?

I'd probably write the same thing as:

    if (typeof gzip$compress === "function") {
        return gzip$compress(data);
    }